### ISSUE TITLE
Showing an overlay widget using the canvas.Overlays().Add() API blocks interaction with rest of window

### ISSUE THREAD

**Issue by dweymouth** [fyne-io/fyne#5051]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

It's unclear if this was intended behavior or not (it's certainly intended for the PopUp case), but the overlay API documentation is just:

```go
// OverlayStack is a stack of CanvasObjects intended to be used as overlays of a Canvas.
type OverlayStack interface {
	// Add adds an overlay on the top of the overlay stack.
	Add(overlay CanvasObject)
	// List returns the overlays currently on the overlay stack.
	List() []CanvasObject
	// Remove removes the given object and all objects above it from the overlay stack.
	Remove(overlay CanvasObject)
	// Top returns the top-most object of the overlay stack.
	Top() CanvasObject
}
```

It says nothing about blocking interaction with what's underneath the overlays, and the documentation suggests you could use it to just show any floating content, which is not the case. If it is intended behavior, the documentation of the OverlayStack should be updated to reflect that, and we should consider introducing a non-blocking variant (#4836). If it isn't intended, we should remove the behavior by default and have some way for overlays like PopUps and Dialogs to indicate that they should block interaction and have their own focus manager.

### How to reproduce

show an overlay using the overlays api over any Fyne app

### Example code

n/a

### Fyne version

develop

### Go compiler version

n/a

### Operating system and version

n/a

**Comment by andydotxyz** [comment]
Yes that is intended - it is for creating new contexts over the current content.

A new "floating" overlay type would be needed to handle things like tooltips. Updating the docs sounds like a good idea.
